src/platform/imx8/platform.c

```diff
@@ -176,6 +176,9 @@ int platform_init(struct sof *sof)
 	sof->platform_dma_domain = dma_multi_chan_domain_init
 			(&sof->dma_info->dma_array[0], 1,
 			 PLATFORM_DEFAULT_CLOCK, false);
+
+	/* i.MX platform DMA domain will be full synchronous, no time dependent */
+	sof->platform_dma_domain->full_sync = true;
```

Comment:
Seems like this should be an argument or flag to dma_multi_chan_domain_init()

Reply:
If you agree to add these changes under, the current, synchronous attribute we won't need to modify the function signature(s).

Reply:
@lgirdwood so one reason for adding a new *mode* (``full_sync``) is that we don't want to force other platforms to use our *optimization*. 
Anyhow, I do think that this might be a real problem on Intel platforms too (although I think none of them uses DMA DAI for scheduling at this moment)

@slawblauciak  @keyonjie can you guys have a look as I've seen you've experimented a little bit with scheduling code.

Reply:
ok, I think this is a topology scheduler type now.